5/README.md

```diff
@@ -45,23 +53,72 @@ There are many use cases for the usage of a Property:
+* If the device is started without a Tango Database, the device and class properties MAY be persisted in a file (but no support for the attribute properties) 
+* All device and class properties MAY be directly modified and accessed through the database device or its file
+* Attribute properties SHOULD be modified and accessed at any time through the device object (RFC-2)
+* All default attribute and device properties are OPTIONAL
+* The attribute, class and device properties MAY be defined in the device code and they MAY be OPTIONAL OR REQUIRED (mandatory)
+* TODO memorized attributes, system properties
+* TODO : properties history
+* TODO: error management
+* All device and class properties MUST be loaded at init command or device start-up
+* If a device property does not exist, the device MUST load the class property having the same name
+* All device and class properties MAY be modified and accessed through the database device when using a Tango Database (RFC-6)
+* A Tango device MUST manage the following default device properties: cmd_min_poll_period, min_poll_period, attr_min_poll_period, poll_ring_depth, cmd_poll_ring_depth, attr_poll_ring_depth, polled_attr, logging_target, logging_level, logging_rft (TODO: description of each one)
+* An attribute MUST manage the following default attribute properties (TODO: descriptions):
+	 * String **label** default value "";
+	 * String **description** default value  "No description"
+	 * String **unit** default value "No unit"
+	 * String **standard_unit** default value "No standard unit"
+	 * String **display_unit** default value "No display unit"
+	 * String **format** default value :
+				* attribute type is a string, "%s"
+				* attribute type is float or double, "%6.2f"
+				* "Not specified" otherwise
+  * String **min_value** default value "Not specified"
+  * String **max_value** default value  "Not specified"
+  * String **min_alarm** default value "Not specified"
+  * String **max_alarm** default value  "Not specified"
+  * String **min_warning** default value  "Not specified"
+  * String **max_warning** default value  "Not specified"
+  * String **delta_t** default value  "Not specified"
+  * String **delta_val** default value "Not specified"
+		* String **abs_change** default value "Not specified"
+		* String **rel_change** default value "Not specified"
+		* String **event_period** default value "Not specified"
+		* String **archive_period** default value "Not specified"
+		* String **archive_rel_change** default value "Not specified"
+		* String **archive_abs_change** default value "Not specified"
+		* String **enum_labels** default value "Not specified"
+		* String **__root_att** default value "Not specified"
+		* String **enum_labels**
+		* String **__value**
+
 
 ### Naming convention
 * The Property's Name SHALL use the following convention:
 ``` ABNF
 property-name = 1*VCHAR
 ```
```

Comment:
From my testing with Jive I've found that I can add a property with an empty name as well. And I can also just use a space (` `) as name. There are even no checks for duplicated names so two properties with empty names are possible as well. So I think this is more like `property-name = 0*CHAR`.

Looking into the database schema, https://github.com/tango-controls/TangoDatabase/blob/master/create_db_tables.sql.in, the name is limited to 255 characters so we can write it as.

```ABNF
property-name = 0*255CHAR
```

Reply:
> From my testing with Jive I've found that I can add a property with an empty name as well. And I can also just use a space (` `) as name. There are even no checks for duplicated names so two properties with empty names are possible as well. So I think this is more like `property-name = 0*CHAR`.

This would not be true if we consider properties defined from a file (without Tango Database).
POGO does some checks on the property name and won't accept an empty property name and would remove any space. POGO also forbids device property names with some special characters like '%' for instance which could be interpreted as a wildcard by a Database engine.

> Looking into the database schema, https://github.com/tango-controls/TangoDatabase/blob/master/create_db_tables.sql.in, the name is limited to 255 characters so we can write it as.

Indeed, there is this limitation by default.
This is also defined in cppTango for instance:
https://github.com/tango-controls/cppTango/blob/tango-9-lts/cppapi/server/tango_const.h.in#L221

> 
> ```abnf
> property-name = 0*255CHAR
> ```

As we said this morning during the teleconf, this might be true with the current jive implementation (not true when using POGO or a file DB). We will make a note that the current implementation allows this kind of property name (empty name of with spaces and tabs) but this is due to issues with the current implementation. empty property names, property names with tabs, spaces and some special characters (we need to find the list) should be forbidden.

Reply:
Here is the method which does the check in POGO:
https://github.com/tango-controls/pogo/blob/Pogo-Release-9.6.23/org.tango.pogo.pogo_gui/src/org/tango/pogo/pogo_gui/tools/Utils.java#L404

As I understand this code, only alphanumeric characters are allowed [a-z][A-Z][0-9] and the underscore.
The property shall start with a letter.
An attribute property (and only an attribute property) can start with an underscore.

@srgblnch , @hardion , this could be useful for #7: this [POGO piece of code](https://github.com/tango-controls/pogo/blob/Pogo-Release-9.6.23/org.tango.pogo.pogo_gui/src/org/tango/pogo/pogo_gui/tools/Utils.java#L404) (combined with a slightly deeper analysis) also highlights that "State" and "Status" command names are reserved. Command names seem to be case sensitive because command names "state" and "status" seem to be accepted by POGO.
I've done a quick test generating a C++ device server with a command named "state" returning a DevState and it seems that this use case is not well supported. Pogo generated the source code, but created a method 
`Tango::DevState MyClassName::state()`

AtkPanel did not recognize this "state" command. tg_devtest (jive test device) listed "state" and "State" as 2 commands available on the device but executing the "state" command from tg_devtest did not execute Tango::DevState MyClassName::state() method, but MyClass::dev_state() was executed instead, as for the "State" command.

Similar behaviour when adding a command named "status".

So it would probably be good to specify that "state" and "status" should be reserved|forbidden too as command names.

Reply:
This is indeed a deficiency of Pogo and should be avoided. To be added to RFC on Commands, Attributes and State.

Reply:
Since the specifications were not there at the time of implementation, I think it is normal that we fall into those kind of inconsistencies. So I think we should compromise on a specification even though the current implementation does not reflect 100% the specification. I propose:
`property-name = 1*255ALPHA`

Reply:
> @srgblnch , @hardion , this could be useful for #7: this [POGO piece of code](https://github.com/tango-controls/pogo/blob/Pogo-Release-9.6.23/org.tango.pogo.pogo_gui/src/org/tango/pogo/pogo_gui/tools/Utils.java#L404) (combined with a slightly deeper analysis) also highlights that "State" and "Status" command names are reserved. Command names seem to be case sensitive because command names "state" and "status" seem to be accepted by POGO.
> I've done a quick test generating a C++ device server with a command named "state" returning a DevState and it seems that this use case is not well supported. Pogo generated the source code, but created a method
> `Tango::DevState MyClassName::state()`

Indeed a good point. The draft of the device RFC already define to have a State and Status attributes. Would it make sense to define what is a State and a Status command in the Command RFC and make mandatory the State and Status command in the Device RFC?

Reply:
> Here is the method which does the check in POGO:
> https://github.com/tango-controls/pogo/blob/Pogo-Release-9.6.23/org.tango.pogo.pogo_gui/src/org/tango/pogo/pogo_gui/tools/Utils.java#L404
> 
> As I understand this code, only alphanumeric characters are allowed [a-z][A-Z][0-9] and the underscore.
> The property name shall start with a letter.
> An attribute property (and only an attribute property) can start with an underscore.

So if we follow the rules defined by code in POGO, I think we would get something like that for the property names in abnf:

```
alphanum = ALPHA / DIGIT 
underscore = %x5F
device_property_name = 1*ALPHA *(alphanum / underscore)
class_property_name = 1*ALPHA *(alphanum / underscore)
free_property_name = 1*ALPHA *(alphanum / underscore)
attribute_property_name = 1*(ALPHA / underscore) *(alphanum / underscore)
```

Reply:
And if we take into account that the maximum property name length is 255 we would get something like:
```
alphanum = ALPHA / DIGIT 
underscore = %x5F
device_property_name = 1*1ALPHA 0*254(alphanum / underscore)
class_property_name = 1*1ALPHA 0*254(alphanum / underscore)
free_property_name = 1*1ALPHA 0*254(alphanum / underscore)
attribute_property_name = 1*1(ALPHA / underscore) 0*254(alphanum / underscore)
```